### ISSUE TITLE
Localised crowdsourced data

### ISSUE THREAD

**Issue by wfdd** [mysociety/yournextrepresentative#677]
Currently, the database is not locale-aware, nor is there a means to submit data in multiple languages.  To facilitate the use of YNR in multilingual jurisdictions, forms should, at a minimum, allow for the submission of data specific to the active user-interface locale.  Things to consider:

* What changes are we gonna have to make to the models?  Would any of [these](https://www.djangopackages.com/grids/g/model-translation/) libraries suit our needs?
* Would we let users submit data in locales other than the active one?
  * If so, how are the forms gonna be laid out?  Should there be multiple input boxes, where applicable?  Should there be separate forms for each of the languages and a tabbed interface to switch between them?
* Would locales be specified at the website or the election level?

**Comment by tfgg** [comment]
I don't quite understand this one - what sort of fields would be in multiple languages? How do you keep data in different locales synchronised? YNR already has the ability to have multiple names for someone.

**Comment by wfdd** [comment]
> I don't quite understand this one - what sort of fields would be in multiple languages?

Everything, from the names of posts, to the names of the candidates, to candidates' biographies.

> How do you keep data in different locales synchronised?

Through manual review, for the most part.

**Comment by akuckartz** [comment]
:+1: (but please use JSON-LD where appropriate :-)

**Comment by tfgg** [comment]
> Everything, from the names of posts, to the names of the candidates, to candidates' biographies.

Names of posts are a separate thing from the other two, since they're not crowd sourced. I note that in the UK, Welsh constituencies have unofficial Welsh names. Do other countries have constituencies with different names in multiple languages?

> Through manual review, for the most part.

This seems to make ensuring completeness and quality of data difficult, which is more important here than, say, wikipedia, but I see.

@akuckartz Could you keep the JSON-LD chat to issue #364 and #590?

**Comment by wfdd** [comment]
> I note that in the UK, Welsh constituencies have unofficial Welsh names. Do other countries have constituencies with different names in multiple languages?

Yes, Cypriot constituencies have official Greek and Turkish names.  (Somewhat tangentially, the election results website is [trilingual](http://live.elections.moi.gov.cy/English/EUROPEAN_ELECTIONS_2014/Islandwide).)

> This seems to make ensuring completeness and quality of data difficult, which is more important here than, say, wikipedia.

It depends.  It'd be no easy task in a place with as many constituencies and candidates as Britain, but I'd wager it's feasible to do in Cyprus (well, or maybe not.... critical mass and all that); I think, it'd primarily rest on the success of the review team (though, admittedly, things are looking pretty bleak in that regard).  In any case, if one language version were to lack information found in another, I don't think that's necessarily an issue.  It could be helped by - say - displaying a notice that language X has been updated more (most) recently.

**Comment by mhl** [comment]
@wfdd We're going to look into this further - we wondered if you have ideas about a preferred implementation yourself? (e.g. any Django package to help that you'd recommend?)

**Comment by wfdd** [comment]
Afraid not.  I could do some research, but I think you've got a lot more experience working with Django.

**Comment by struan** [comment]
I note the [popit_ng](https://github.com/Sinar/popit_ng) uses https://github.com/KristianOellegaard/django-hvad for translating fields and at least it looks nice in the code - https://github.com/Sinar/popit_ng/blob/master/popit/models/person.py#L20
